javalib/src/main/scala/java/util/Arrays.scala

```diff
@@ -625,37 +639,46 @@ object Arrays {
   }
 
   @noinline def hashCode(a: Array[Long]): Int =
-    hashCodeImpl[Long](a)
+    hashCodeImpl[Long](a, _.hashCode())
 
   @noinline def hashCode(a: Array[Int]): Int =
-    hashCodeImpl[Int](a)
+    hashCodeImpl[Int](a, _.hashCode())
 
   @noinline def hashCode(a: Array[Short]): Int =
-    hashCodeImpl[Short](a)
+    hashCodeImpl[Short](a, _.hashCode())
 
   @noinline def hashCode(a: Array[Char]): Int =
-    hashCodeImpl[Char](a)
+    hashCodeImpl[Char](a, _.hashCode())
 
   @noinline def hashCode(a: Array[Byte]): Int =
-    hashCodeImpl[Byte](a)
+    hashCodeImpl[Byte](a, _.hashCode())
 
   @noinline def hashCode(a: Array[Boolean]): Int =
-    hashCodeImpl[Boolean](a)
+    hashCodeImpl[Boolean](a, _.hashCode())
 
   @noinline def hashCode(a: Array[Float]): Int =
-    hashCodeImpl[Float](a)
+    hashCodeImpl[Float](a, _.hashCode())
 
   @noinline def hashCode(a: Array[Double]): Int =
-    hashCodeImpl[Double](a)
+    hashCodeImpl[Double](a, _.hashCode())
 
   @noinline def hashCode(a: Array[AnyRef]): Int =
     hashCodeImpl[AnyRef](a, Objects.hashCode(_))
 
   @inline
-  private def hashCodeImpl[T](a: Array[T],
-      elementHashCode: T => Int = (x: T) => x.hashCode): Int = {
-    if (a == null) 0
-    else a.foldLeft(1)((acc, x) => 31*acc + elementHashCode(x))
+  private def hashCodeImpl[T](a: Array[T], elementHashCode: T => Int): Int = {
+    if (a == null) {
+      0
+    } else {
+      var acc = 1
+      val len = a.length
+      var i = 0
+      while (i != len) {
+        acc = 31*acc + elementHashCode(a(i))
+        i += 1
+      }
```

Comment:
Can't this be a `for (i <- 0 until a.length)` (with the `ScalaOps` you introduce in the `-Yno-predef` commit?)